tests/cpp/test_multidevice_overlap.cpp

```diff
@@ -472,4 +472,136 @@ TEST_F(
   }
 }
 
+class AGOverlapTest : public MultiDeviceTest {
+ protected:
+  OverlapTestParams params;
+
+  int64_t num_devices_;
+  int64_t my_device_index_;
+  std::vector<int64_t> all_devices_;
+  at::Tensor ta_, tb_, tc_;
+  at::Tensor ta_sharded_, tc_sharded_;
+  // stores the backend
+  c10d::Backend* world_communicator_;
+
+  // Define I/O and intermediate Tensor shapes
+  // A has shape (S, sharded(n), M/(S*n), K)
+  // B(K,N)
+  // C has shape (S, M/S, N)
```

Comment:
This doesn't look right. I believe `C` should be sharded as well.

Reply:
I don't follow you. But this comment is at least different from [the code](https://github.com/NVIDIA/Fuser/pull/3064/files#diff-30df6421558f87ef0024b01f11752c35d3d68b80a9e6e0ec0fd49de535acb91aR521-R522). It would make sense if it says `(S, sharded(n), M/(S*n), N)`. Am I missing something?

Reply:
I'll read it again. I've been incorrectly assuming B is the tensor to allgather and A and C are sharded.